shims/spark310/src/main/scala/com/nvidia/spark/rapids/shims/spark310/GpuBroadcastHashJoinExec.scala

```diff
@@ -109,6 +109,10 @@ case class GpuBroadcastHashJoinExec(
     "joinTime" -> SQLMetrics.createNanoTimingMetric(sparkContext, "join time"),
     "filterTime" -> SQLMetrics.createNanoTimingMetric(sparkContext, "filter time"))
 
+  // Spark 3.1.0 started introducing ordering guarantees for hash joins but GPU
+  // hash joins do not provide ordering guarantees
+  override def outputOrdering: Seq[SortOrder] = Nil
```

Comment:
Is it OK for us to replace a node using stricter ordering guarantees with one that is less conformant?  I don't remember whether the ordering constraint pass has occurred before or after the plugin modifies the plan.

Reply:
The ordering fixup happens after we replace the plan.

https://github.com/NVIDIA/spark-rapids/blob/4cb4f5bd01043828953762ebfeec5e6db8c57769/sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala#L2278-L2279

Reply:
Thanks, @revans2!